racket/collects/syntax/parse/lib/function-header.rkt

```diff
@@ -49,63 +48,25 @@
            #:attr default #f)
   (pattern (~seq kw:keyword [name:id default])))
 
-;; invalid-option-placement : (Listof Id) (Listof Syntax/#f) -> Id/#f
+;; invalid-option-placement : (Listof Keyword) (Listof Id) (Listof Syntax/#f) -> Id/#f
 ;; Checks for mandatory argument after optional argument; if found, returns
 ;; identifier of mandatory argument.
-(define (invalid-option-placement names defaults)
-  ;; find-mandatory : (Listof Id) (Listof Syntax/#f) -> Id/#f
+(define (invalid-option-placement kws names defaults)
+  ;; find-mandatory : (Listof Keyword) (Listof Id) (Listof Syntax/#f) -> Id/#f
   ;; Finds first name w/o corresponding default.
-  (define (find-mandatory names defaults)
-    (for/first ([name (in-list names)]
+  (define (find-mandatory kws names defaults)
+    (for/first ([kw (in-list kws)]
+                [name (in-list names)]
                 [default (in-list defaults)]
-                #:when (not default))
+                #:when (and (not kw) (not default)))
       name))
   ;; Skip through mandatory args until first optional found, then search
   ;; for another mandatory.
-  (let loop ([names names] [defaults defaults])
+  (let loop ([kws kws] [names names] [defaults defaults])
     (cond [(or (null? names) (null? defaults))
            #f]
-          [(eq? (car defaults) #f) ;; mandatory
-           (loop (cdr names) (cdr defaults))]
+          [(or (car kws) ;; keyword
```

Comment:
This is probably not correct, is it?
If it is, can you add a test case that would break without it?

Reply:
Oh I see.
Here are 2 test cases:
This one should pass: `(a #:b [b 1] c)`
This one should fail with `"default value missing"`: `(a [b 1] c)`